tests/test_apis.py

```diff
@@ -2,19 +2,42 @@
 
 from __future__ import unicode_literals
 import unittest
+import socket
 
 import dotdot
 
 from pubs.p3 import ustr
 from pubs.endecoder import EnDecoder
-from pubs.apis import doi2bibtex, isbn2bibtex
+from pubs.apis import ReferenceNotFoundError, arxiv2bibtex, doi2bibtex, isbn2bibtex, _is_arxiv_oldstyle, _extract_arxiv_id
 
+from pubs import apis
 
-class TestDOI2Bibtex(unittest.TestCase):
+
+def _is_connected():
+    """Return False if no internet connection is detected.
+
+    May not work if the local network redirects the connection.
+    """
+    try:
+        host = socket.gethostbyname('www.google.com')
+        s = socket.create_connection((host, 80), 2)
+        s.close()
+        return True
+    except socket.error:
+        pass
+    return False
+
+class APITests(unittest.TestCase):
 
     def setUp(self):
+        if not _is_connected():
+            self.skipTest('no connection detected, skiping test')
```

Comment:
Can't we use a `@unittest.skipIf(not _is_connected)` instead? This would make it more explicit. And we could also have tests in the API that don't use the network. That's a good way to do it too, so it's fine anyway if you like it more.

Reply:
With the mock properly implemented, not needed anymore.